### PR TITLE
doc (jkube-kit/doc) : Add documentation for Helidon framework support (#1475)

### PR DESCRIPTION
**Pull request by rohanKanojia**
## Description

:heavy_check_mark: ~Requires #2129 #2130 to be merged first~

+ Add documentation for HelidonGenerator
+ Add documentation for HelidonHealthCheckEnricher

## Type of change

 - [ ] Bug fix (non-breaking change which fixes an issue)
 - [ ] Feature (non-breaking change which adds functionality)
 - [ ] Breaking change (fix or feature that would cause existing functionality to change
 - [X] Chore (non-breaking change which doesn't affect codebase;
   test, version modification, documentation, etc.)

## Checklist
 - [X] I have read the [contributing guidelines](https://www.eclipse.org/jkube/contributing)
 - [X] I signed-off my commit with a user that has signed the [Eclipse Contributor Agreement](https://www.eclipse.org/legal/ECA.php)
 - [X] My code follows the style guidelines of this project
 - [X] I Keep It Small and Simple: The smaller the PR is, the easier it is to review and have it merged
 - [X] I use [conventional commits](https://www.conventionalcommits.org/) in my commit messages
 - [X] I have performed a self-review of my code
 - [ ] I Added [CHANGELOG](../CHANGELOG.md) entry
 - [ ] I have updated the [documentation](../kubernetes-maven-plugin/doc) accordingly
 - [ ] No new bugs, code smells, etc. in [SonarCloud](https://sonarcloud.io/dashboard?id=jkubeio_jkube) report
 - [ ] I have added tests that prove my fix is effective or that my feature works
 - [ ] New and existing unit tests pass locally with my changes
 - [ ] I tested my code in Kubernetes
 - [ ] I tested my code in OpenShift

### CODE REVIEW

File: jkube-kit/doc/src/main/asciidoc/inc/_generator.adoc [hunk 1]
```diff
@@ -71,11 +75,12 @@ include::generator/_webapp.adoc[]
 include::generator/_quarkus.adoc[]
 include::generator/_openliberty.adoc[]
 include::generator/_micronaut.adoc[]
-
-include::generator/_api.adoc[]
 ifeval::["{plugin-type}" == "maven"]
 include::generator/_karaf.adoc[]
 include::generator/_wildfly_jar.adoc[]
+include::generator/_helidon.adoc[]
```
Comment (manusa) [inline jkube-kit/doc/src/main/asciidoc/inc/_generator.adoc:81]: Doesn't the generator work on Gradle?

Comment (rohanKanojia) [reply jkube-kit/doc/src/main/asciidoc/inc/_generator.adoc:81]: Helidon doesn't support Native Image mode in gradle.

Comment (manusa) [reply jkube-kit/doc/src/main/asciidoc/inc/_generator.adoc:81]: When I reviewed your PRs from yesterday, I saw that there were tests that contained a similar project to: https://github.com/helidon-io/helidon/blob/2.6.0/examples/quickstarts/helidon-quickstart-mp/build.gradle

The generator doc contains entries for gradle (within this same PR). So again, I'm not sure if we are supporting or not Gradle. If not, then everything is pretty inconsistent.

> Even [Helidon Starter](https://helidon.io/starter/3.2.1?step=1) support maven only at the moment.

This shouldn't be relevant. It should be relevant if Helidon works on Gradle or not.
- https://github.com/helidon-io/helidon/blob/a20fe937749cf1b42701a24b7de4ad941b738cba/examples/quickstarts/helidon-quickstart-se/build.gradle
- https://github.com/helidon-io/helidon/blob/a20fe937749cf1b42701a24b7de4ad941b738cba/examples/quickstarts/helidon-quickstart-mp/build.gradle

Have you tried these? could you clarify if they work or not.

Comment (manusa) [reply jkube-kit/doc/src/main/asciidoc/inc/_generator.adoc:81]: Ok, but it does work in standard JVM mode, so we should add it nonetheless
Also we need to keep this in mind. We can have a better UX by warning users that Native mode is not applicable in gradle. so probably an improvement for the generator

Comment (rohanKanojia) [reply jkube-kit/doc/src/main/asciidoc/inc/_generator.adoc]: I've moved generator and healthcheck enricher to gradle

**Review by manusa — APPROVED** [review]

**Pull request merged**